### PR TITLE
[NONEVM-904] Update address book to support nonevm

### PR DESCRIPTION
**Pull request by jlaveracll**

### CODE REVIEW

File: deployment/address_book.go [hunk 1]
```diff
@@ -93,20 +93,22 @@ type AddressBookMap struct {
...
 	}
-	if address == "" || address == common.HexToAddress("0x0").Hex() {
-		return errors.Wrap(ErrInvalidAddress, "address cannot be empty")
-	}
-	if common.IsHexAddress(address) {
-		// IMPORTANT: WE ALWAYS STANDARDIZE ETHEREUM ADDRESS STRINGS TO EIP55
-		address = common.HexToAddress(address).Hex()
-	} else {
-		return errors.Wrapf(ErrInvalidAddress, "address %s is not a valid Ethereum address, only Ethereum addresses supported", address)
+	if family == chainsel.FamilyEVM {
```
Comment (connorwstein) [inline deployment/address_book.go:102]: could add your aptos specific validation?

File: deployment/address_book.go [hunk 2]
```diff
@@ -93,20 +93,22 @@ type AddressBookMap struct {
 	mtx              sync.RWMutex
 }
 
-// save will save an address for a given chain selector. It will error if there is a conflicting existing address.
-func (m *AddressBookMap) save(chainSelector uint64, address string, typeAndVersion TypeAndVersion) error {
-	_, exists := chainsel.ChainBySelector(chainSelector)
-	if !exists {
+// Save will save an address for a given chain selector. It will error if there is a conflicting existing address.
+func (m *AddressBookMap) Save(chainSelector uint64, address string, typeAndVersion TypeAndVersion) error {
```
Comment (connorwstein) [inline deployment/address_book.go]: Let's update the Save test with some other family addresses

**Review by connorwstein — COMMENTED** [review]
A large number of the comments are a little redundant because they are in EVM specific contexts but fine just keep em and over-communicate imo.

File: deployment/address_book.go (hunk 2 above)
Comment (krehermann) [reply deployment/address_book.go]: why did this become public? i don't see anything that using it in other packages.

File: deployment/address_book.go (hunk 1 above)
Comment (jlaveracll) [reply deployment/address_book.go:102]: we don't have any yet, but we'll create a ticket to add it later. We are already tackling hardening tasks like this one

File: deployment/address_book.go (hunk 2 above)
Comment (jlaveracll) [reply deployment/address_book.go]: case added!

File: deployment/address_book.go (hunk 1 above)
Comment (cgruber) [reply deployment/address_book.go:102]: I was toying with an idea of building in aptos and other chain or family validations into the chainlink-selectors repo, so we'd have a set of reusable validations keyed off of chain ID in a way that hides all the conditionality from calling code. Valid addresses was definitely in the right zone.  Doing it here is fine, but we should think about making this more generally usable.

**Review by jmank88 — DISMISSED** [review]

File: core/services/relay/evm/evm.go [hunk 3]
```diff
@@ -197,6 +197,7 @@ func NewRelayer(ctx context.Context, lggr logger.Logger, chain legacyevm.Chain,
 	sugared := logger.Sugared(lggr).Named("Relayer")
 	mercuryORM := mercury.NewORM(opts.DS)
 	cdcFactory := sync.OnceValues(func() (llo.ChannelDefinitionCacheFactory, error) {
+		// NOTE: This does not support non-evm chains
```
Comment (cgruber) [inline core/services/relay/evm/evm.go]: Are these notes intended to be TODOs, or is this purely documentary?

File: deployment/address_book.go (hunk 2 above)
Comment (connorwstein) [reply deployment/address_book.go]: yeah +1 @krehermann this should remain a private helper, there's already a public Save. Not sure how this actually compiles it looks like it has the same signature as the public Save?

**Review by cgruber — DISMISSED** [review]

**Review by connorwstein — DISMISSED** [review]

**Review by jmank88 — DISMISSED** [review]

File: core/services/relay/evm/evm.go (hunk 3 above)
Comment (jlaveracll) [reply core/services/relay/evm/evm.go]: documentation

**Review by jmank88 — DISMISSED** [review]

**Review by cgruber — DISMISSED** [review]

**Review by connorwstein — DISMISSED** [review]

**Review by krehermann — DISMISSED** [review]

**Review by connorwstein — DISMISSED** [review]

**Review by prashantkumar1982 — APPROVED** [review]

**Review by prashantkumar1982 — DISMISSED** [review]

**Review by prashantkumar1982 — DISMISSED** [review]

**Review by krehermann — DISMISSED** [review]

**Review by connorwstein — APPROVED** [review]

**Review by krehermann — APPROVED** [review]

**Review by krehermann — APPROVED** [review]

**Review by prashantkumar1982 — APPROVED** [review]

**Pull request merged**